File: leoz-ui/src/main/java/org/deku/leoz/Global.java
package org.deku.leoz;

/**
 * Leoz global resources
 *
 * Created by masc on 22.09.14.
 */
public class Global {
    private static Global mInstance;

    public static Global instance() {
        if (mInstance == null) {
            synchronized(Global.class) {
                mInstance = new Global();
            }
        }
        return mInstance;
    }

    private Global() {
    }
}

File: leoz-ui/src/main/java/org/deku/leoz/Global.kt
package org.deku.leoz

/**
 * Leoz global resources

 * Created by masc on 22.09.14.
 */
object Global {
    private var mInstance: Global? = null

    fun instance(): Global? {
        if (mInstance == null) {
            synchronized (Global::class.java) {
                mInstance = Global()
            }
        }
        return mInstance
    }
}
